app/controllers/companies_controller.rb

```diff
@@ -4,9 +4,15 @@ class CompaniesController < ApplicationController
   load_and_authorize_resource
 
   def index
-    companies = current_user.company ? Company.accessible_by(current_ability) : Company.all
-
+    offset_page = page
+    companies = if current_user.company
+                  Company.accessible_by(current_ability).offset(offset_page).limit(default_page_size)
```

Comment:
может  offset(offset_page).limit(default_page_size) вынести в метод который будет работать с коллекцией ?
def paginate_collection(collection)
   collection.offset(offset_page).limit(default_page_size)
end


```suggestion
                  paginate_collection(Company.accessible_by(current_ability))
```